### PR TITLE
wrap AliasDb in Python

### PR DESCRIPTION
**Pull request by t-vi**
Also added a wrapper @tlemo 's graphviz export to string.

### CODE REVIEW

**Comment by t-vi** [comment]
I'm unsure what to do about the clang-tidy. It complains about the GS macro that is in the "context" bits of my patch, but it isn't changed or used by me.

**Review by eellison — APPROVED** [review]
Nice!

**Comment by eellison** [comment]
Most of the clang tidy errors are noise...

**Review by tlemo — APPROVED** [review]
Looks great to me!

PS. @eellison, do we still want the binding in https://github.com/pytorch/pytorch/pull/50969 considering this PR?

File: torch/csrc/jit/python/python_ir.cpp [hunk 1]
```diff
@@ -209,11 +209,22 @@ Node* Graph::createPythonOp(
 
 void initPythonIRBindings(PyObject* module_) {
   auto m = py::handle(module_).cast<py::module>();
+
+  py::class_<AliasDb, std::shared_ptr<AliasDb>>(m, "AliasDb")
+      .def("dump", &AliasDb::dump)
+      .def("to_graphviz_str", &AliasDb::toGraphviz)
```
[the remark this answers is not in the document]
Comment (t-vi) [reply torch/csrc/jit/python/python_ir.cpp:215]: I wanted to emphasize that the result is a str not a graphviz object. We might yet add a thing returning a graphviz object directly.

File: test/jit/test_python_bindings.py [hunk 2]
```diff
@@ -35,3 +35,13 @@ def fn(x: torch.Tensor):
...
+
+    def test_aliasdb(self):
+        @torch.jit.script
+        def test_aliasdb_fn(x: torch.Tensor):
+            return 2 * x
+
+        gr = test_aliasdb_fn.graph.copy()
+        alias_db = gr.alias_db()
+        self.assertTrue("WILDCARD" in str(alias_db))
+        self.assertTrue("digraph fusion_ir" in alias_db.to_graphviz_str())
```
[the remark this answers is not in the document]
Comment (t-vi) [reply test/jit/test_python_bindings.py:47]: Sure!

**Comment by t-vi** [comment]
Yes, thank you for the reminder!

**Comment by t-vi** [comment]
@eellison So the failing clang-tidy is from the diff context, not the PR. Anything I can do to move it along?

**Comment by eellison** [comment]
@t-vi I ignore clang tidy lol, no need to worry about that. There were just some internal flakey errors yesterday, trying to land today.

**Pull request closed without merging**